### DISCUSSION TITLE
Application keeps forgetting pump circuits

### DISCUSSION THREAD

**Discussion by jslabotsky** [tagyoureit/nodejs-poolController#460] [App and O/S Config Questions] [answered]
I just updated to latest version. Every time the controller is restarted, all of the circuits disappear from the pump configuration and have to be re-added.

Separate topic, but I also notice that it only sends updates to the wattage of my Intelliflo VS pump when the speed is changed. It used to update in real time while it is running. In Dashpanel the watts readout does change periodically, but I have to manually refresh the page.

**Comment by rstrouse** [comment]
Hmmm...  That sounds like the socket connection is not alive between dashPanel and njsPC.  Can I get you to upload a replay so I can replicate your environment?

**Reply by jslabotsky** [reply]
Thanks, here you go. Hopefully I did it right.
[replay.zip](https://github.com/tagyoureit/nodejs-poolController/files/8764555/replay.zip)

**Comment by rstrouse** [comment, marked as answer]
Pull njsPC the ids for Nixie pump circuits were not being set so the cleanup recovery code was clearing them out on startup.

**Reply by jslabotsky** [reply]
That did the trick. Thank you!!!

**Reply by DrJeff** [reply]
@rstrouse I had the same issue thought my Pi was having issues, pulled the newest now thanks.

**Reply by rstrouse** [reply]
Yeah the indexing of the circuits was not enough.  These needed ids applied to them.
